src/gurt/misc.c

```diff
@@ -854,6 +858,8 @@ d_rank_range_list_create_from_ranks(d_rank_list_t *rank_list)
 	unsigned int			i;		/* rank index */
 	unsigned int			j;		/* rank range index */
 
+	D_ASSERT(rank_list != NULL);
```

Comment:
Not saying this assertion is wrong, but the `rank_list == NULL` on new line 864 now looks weird---something isn't quite consistent here.

Reply:
- [ ] Remove invalid ASSERT
- [ ] Update unit test